traffic_portal/app/src/common/api/CDNService.js

```diff
@@ -17,252 +17,207 @@
  * under the License.
  */
 
-var CDNService = function($http, $q, Restangular, locationUtils, messageModel, ENV) {
+var CDNService = function($http, locationUtils, messageModel, ENV) {
 
     this.getCDNs = function(all) {
-        var request = $q.defer();
-
-        $http.get(ENV.api['root'] + "cdns")
-            .then(
-                function(result) {
-                    var response;
-                    if (all) { // there is a CDN called "ALL" that is not really a CDN but you might want it...
-                        response = result.data.response;
-                    } else {
-                        response = _.filter(result.data.response, function(cdn) {
-                            return cdn.name != 'ALL';
-                        });
-                    }
-                    request.resolve(response);
-                },
-                function(fault) {
-                    request.reject();
+        return $http.get(ENV.api['root'] + 'cdns').then(
```

Comment:
so i'm seeing some weird errors and i think it's because of the way the angular-ui-router's "resolves" work. for example, if you try to go here (when you are unauthorized/logged in) - https://localhost:8442/#!/cdns - the way I built TP with the help of angular-ui-router is it works its way down the resolve chain.

it first tries to resolve these resolves:

https://github.com/apache/trafficcontrol/blob/master/traffic_portal/app/src/modules/private/index.js#L45

specifically, it first tries to fetch "current user" and if that fails, the next resolve should NOT even be attempted:

https://github.com/apache/trafficcontrol/blob/master/traffic_portal/app/src/modules/private/cdns/list/index.js#L29

but with your changes (removing $q.defer()), it's not stopping after the first resolve failure...instead it just keeps going down the resolve chain. maybe that's not a problem...but maybe it is. will need to do some more testing. 

but i do think this format works and it has no restangular (your goal of this PR)

`    this.getCDNs = function(all) {
        var request = $q.defer();
        $http.get(ENV.api['root'] + "cdns")
            .then(
                function(result) {
                    var response;
                    if (all) { // there is a CDN called "ALL" that is not really a CDN but you might want it...
                        response = result.data.response;
                    } else {
                        response = _.filter(result.data.response, function(cdn) {
                            return cdn.name != 'ALL';
                        });
                    }
                    request.resolve(response);
                },
                function(fault) {
                    request.reject(fault);
                }
            );

        return request.promise;
    };
`

Reply:
UserService.js#getCurrentUser() needs to be implemented like this (or something with similar behavior) so if it fails, the remaining resolves are not attempted at all:

```
    this.getCurrentUser = function() {
        const deferred = $q.defer();
        $http.get(ENV.api['root'] + "user/current")
            .then(
                function(result) {
                    userModel.setUser(result.data.response);
                    deferred.resolve(result.data.response);
                },
                function(fault) {
                    deferred.reject(fault);
                }
            );
        return deferred.promise;
    };
```

Reply:
^^ you can disregard all of that. see the comment attached to UserService.js